scripts/grafana_pdf.py

```diff
@@ -1,41 +1,108 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
 
 from __future__ import print_function, \
     unicode_literals
 
+import argparse
 import os
-import urllib2
+import time
 import json
 import tarfile
 import shutil
 
+try:
+    # For Python 2
+    import urllib2 as urlreq
+    from urllib2 import HTTPError, URLError
+except ImportError:
+    # For Python 3
+    import urllib.request as urlreq
+    from urllib.error import HTTPError, URLError
+
 dests = []
 download_dir = "grafana_pdf"
 
 if not dests:
     with open("./dests.json") as fp:
         dests = json.load(fp)
 
+
 def make_tarfile(output_filename, source_dir):
     with tarfile.open(output_filename, "w:gz") as tar:
         tar.add(source_dir, arcname=os.path.basename(source_dir))
 
-if __name__ == '__main__':
 
+def read_url(url):
+    try:
+        f = urlreq.urlopen(url)
+        return f.read()
+    except HTTPError as e:
+        print("HTTP Error: %s" % e.getcode())
+        return e.read()
+    except URLError as e:
+        print("Reading URL %s error: %s" % (url, e))
+        return None
+
+
+def parse_opts():
+    parser = argparse.ArgumentParser(
+        description="Export Grafana charts to PDF")
+    parser.add_argument("-t", "--time", action="store", default=None,
+                        help="Relative time to now, supported format is like: 2h, 4h. If not set, assume 3h by default.")
+    parser.add_argument("--time-from", action="store", default=None,
+                        help="Start timestamp of time range, format: '%%Y-%%m-%%d %%H:%%M:%%S'.")
+    parser.add_argument("--time-to", action="store", default=None,
+                        help="End timestamp of time range, format: '%%Y-%%m-%%d %%H:%%M:%%S'.")
+    return parser.parse_args()
+
+
+def parse_timestamp(time_string):
+    format_guess = [
+        "%Y-%m-%d %H:%M:%S",
+        "%Y-%m-%d %H:%M",
+        "%Y-%m-%d %H",
+        "%Y-%m-%d",
+        "%m-%d",
+        "%H:%M:%S",
+        "%H:%M",
+        "%H"
+    ]
+    for time_format in format_guess:
+        try:
+            return time.mktime(time.strptime(time_string, time_format))
+        except ValueError:
+            pass
+    raise ValueError(
+        "time data '%s' does not match any supported format." % time_string)
+
+
+if __name__ == '__main__':
+    args = parse_opts()
     if not os.path.isdir(download_dir):
         os.makedirs(download_dir)
 
+    if args.time:
+        time_args = "&from=now-{0}&to=now".format(args.time)
```

Comment:
"&from=now-{0}h&to=now"

Reply:
It's possible to use other time units like `30m`, `1d`, etc.